### PR TITLE
Add HISTORY.md entry for change noted in #39589

### PR DESCRIPTION
**Pull request by sostock**
As noted in https://github.com/JuliaLang/julia/issues/39589#issuecomment-776389595, it could be good to mention this change in NEWS/HISTORY. This adds a HISTORY.md item. I will also make a PR backporting this to the 1.6 NEWS.md file.

Edit: backport PR: #39785

### CODE REVIEW

**Comment by timholy** [comment]
I think it would be best to frame this as a bugfix from the beginning. Phrases like "now always describe" and "is considered a bugfix" make it sound like a bit of a dicey change to be making in the middle of a stable release cycle. But bugfixes can always break someone's code (I won't bother to link the obligatory xkcd on the topic). But it's just a bugfix, pure and simple, no excuses needed.

File: HISTORY.md [hunk 1]
```diff
@@ -163,6 +163,10 @@ Standard library changes
   results table ([#38042]).
 * `@testset` now supports the option `verbose` to show the test result summary
   of the children even if they all pass ([#33755]).
+* In `LinearIndices(::Tuple)` and `CartesianIndices(::Tuple)`, integer elements (as opposed to ranges of integers) of the
+  argument tuple now always describe 1-based ranges, e.g, `CartesianIndices((3, 1:3))` is equivalent to
+  `CartesianIndices((1:3, 1:3))`. This is consistent with the (old and new) behavior for tuples containing only integers
+  and is considered a bugfix ([#37829], [#37928]).
```
Comment (mbauman) [inline HISTORY.md:169]: How about this?  It's a little tricky to _both_ keep with the style of the NEWS (leading with the new behavior) and also lead with the bugfix (which — by definition — leads with the old behavior).  Here's a shot at walking down the middle:

```suggestion
* In `LinearIndices(::Tuple)` and `CartesianIndices(::Tuple)`, integers (as opposed to ranges of integers) in the
  argument tuple now consistently describe 1-based ranges, e.g, `CartesianIndices((3, 1:3))` is equivalent to
  `CartesianIndices((1:3, 1:3))`. This is how tuples of integers have always been documented to work, but a
  bug had caused erroneous behaviors with heterogeneous tuples containing both integers and ranges ([#37829], [#37928]).
```

Comment (sostock) [reply HISTORY.md:169]: I like it.

**Review by mbauman — APPROVED** [review]

**Comment by sostock** [comment]
This has been backported to 1.6-rc3, so this PR can be merged now.

**Comment by mbauman** [comment]
572836ad82ae4b0ef792930f9772cc0dfddb8823

**Pull request merged**